File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/http/HttpClient.php
<?php 
namespace io\vertx\php\core\http;
use io\vertx\php\core\MultiMap;
use io\vertx\php\core\streams\ReadStream;
class HttpClient
{
    
    private function __construct()
    {

    }
    /**
     * ()
     * @return Void
     * methodSize : 1
     */
    public function close(/*0*/)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function delete(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function deleteAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function get(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function getAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (RequestOptions options, callable responseHandler)
     * @return HttpClient
     * (string requestURI, callable responseHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * methodSize : 4
     */
    public function getNow(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function head(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function headAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (RequestOptions options, callable responseHandler)
     * @return HttpClient
     * (string requestURI, callable responseHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * methodSize : 4
     */
    public function headNow(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isMetricsEnabled(/*0*/)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function options(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function optionsAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (RequestOptions options, callable responseHandler)
     * @return HttpClient
     * (string requestURI, callable responseHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClient
     * methodSize : 4
     */
    public function optionsNow(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function post(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function postAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (RequestOptions options)
     * @return HttpClientRequest
     * (string requestURI)
     * @return HttpClientRequest
     * (int port, string host, string requestURI)
     * @return HttpClientRequest
     * (string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (string host, string requestURI)
     * @return HttpClientRequest
     * (RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function put(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (string absoluteURI)
     * @return HttpClientRequest
     * (string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function putAbs(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (callable handler)
     * @return HttpClient
     * methodSize : 1
     */
    public function redirectHandler(/*1*/$arg0)
    {

    }
    /**
     * (enum method, RequestOptions options)
     * @return HttpClientRequest
     * (enum method, string requestURI)
     * @return HttpClientRequest
     * (enum method, int port, string host, string requestURI)
     * @return HttpClientRequest
     * (enum method, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (enum method, string host, string requestURI)
     * @return HttpClientRequest
     * (enum method, RequestOptions options, callable responseHandler)
     * @return HttpClientRequest
     * (enum method, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * (enum method, int port, string host, string requestURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 8
     */
    public function request(/*5*/$arg0, $arg1, $arg2, $arg3, $arg4)
    {

    }
    /**
     * (enum method, string absoluteURI)
     * @return HttpClientRequest
     * (enum method, string absoluteURI, callable responseHandler)
     * @return HttpClientRequest
     * methodSize : 2
     */
    public function requestAbs(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * (RequestOptions options, callable wsConnect)
     * @return HttpClient
     * (string requestURI, callable wsConnect)
     * @return HttpClient
     * (int port, string host, string requestURI, callable wsConnect)
     * @return HttpClient
     * (string host, string requestURI, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, callable wsConnect)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, enum version, callable wsConnect)
     * @return HttpClient
     * (string requestURI, MultiMap headers, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string requestURI, MultiMap headers, enum version, callable wsConnect)
     * @return HttpClient
     * (RequestOptions options, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string host, string requestURI, callable wsConnect)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, callable wsConnect)
     * @return HttpClient
     * (string requestURI, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string requestURI, MultiMap headers, callable wsConnect)
     * @return HttpClient
     * (int port, string host, string requestURI, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, callable wsConnect)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, enum version, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, enum version, callable wsConnect)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, enum version, string subProtocols, callable wsConnect)
     * @return HttpClient
     * (string requestURI, MultiMap headers, enum version, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, enum version, callable wsConnect)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, enum version, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (RequestOptions options, MultiMap headers, enum version, string subProtocols, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect)
     * @return HttpClient
     * (string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, enum version, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect)
     * @return HttpClient
     * (string host, string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * (int port, string host, string requestURI, MultiMap headers, enum version, string subProtocols, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * methodSize : 32
     */
    public function websocket(/*8*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5, $arg6, $arg7)
    {

    }
    /**
     * (string url, MultiMap headers, enum version, string subProtocols, callable wsConnect, callable failureHandler)
     * @return HttpClient
     * methodSize : 1
     */
    public function websocketAbs(/*6*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5)
    {

    }
    /**
     * (RequestOptions options)
     * @return ReadStream
     * (string requestURI)
     * @return ReadStream
     * (int port, string host, string requestURI)
     * @return ReadStream
     * (string host, string requestURI, MultiMap headers)
     * @return ReadStream
     * (RequestOptions options, MultiMap headers, enum version)
     * @return ReadStream
     * (string requestURI, MultiMap headers, enum version)
     * @return ReadStream
     * (string host, string requestURI)
     * @return ReadStream
     * (RequestOptions options, MultiMap headers)
     * @return ReadStream
     * (string requestURI, MultiMap headers)
     * @return ReadStream
     * (int port, string host, string requestURI, MultiMap headers)
     * @return ReadStream
     * (string host, string requestURI, MultiMap headers, enum version)
     * @return ReadStream
     * (RequestOptions options, MultiMap headers, enum version, string subProtocols)
     * @return ReadStream
     * (string requestURI, MultiMap headers, enum version, string subProtocols)
     * @return ReadStream
     * (int port, string host, string requestURI, MultiMap headers, enum version)
     * @return ReadStream
     * (string host, string requestURI, MultiMap headers, enum version, string subProtocols)
     * @return ReadStream
     * (int port, string host, string requestURI, MultiMap headers, enum version, string subProtocols)
     * @return ReadStream
     * methodSize : 16
     */
    public function websocketStream(/*6*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5)
    {

    }
    /**
     * (string url, MultiMap headers, enum version, string subProtocols)
     * @return ReadStream
     * methodSize : 1
     */
    public function websocketStreamAbs(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/http/HttpClient.java
package io.vertx.php.core.http;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.http.HttpClientRequest;
import io.vertx.php.core.http.HttpClientResponse;
import io.vertx.php.core.metrics.Measured;
import io.vertx.php.core.MultiMap;
import io.vertx.php.core.streams.ReadStream;
import io.vertx.php.core.http.WebSocket;
import io.vertx.php.core.Future;
@Name("HttpClient")
@Namespace("io\\vertx\\php\\core\\http")
public interface HttpClient<S extends io.vertx.core.http.HttpClient> extends IWrapper<S>{
    
    @Signature
    default Memory isMetricsEnabled(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isMetricsEnabled()
    }

    @Signature
    default Memory request(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().request(method, options)
    }

    @Signature
    default Memory requestAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().requestAbs(method, absoluteURI)
    }

    @Signature
    default Memory get(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().get(options)
    }

    @Signature
    default Memory getAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().getAbs(absoluteURI)
    }

    @Signature
    default Memory getNow(Environment __env__, Memory... args) {
        this.getWrappedObject().getNow(options, responseHandler)
    }

    @Signature
    default Memory post(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().post(options)
    }

    @Signature
    default Memory postAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().postAbs(absoluteURI)
    }

    @Signature
    default Memory head(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().head(options)
    }

    @Signature
    default Memory headAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().headAbs(absoluteURI)
    }

    @Signature
    default Memory headNow(Environment __env__, Memory... args) {
        this.getWrappedObject().headNow(options, responseHandler)
    }

    @Signature
    default Memory options(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().options(options)
    }

    @Signature
    default Memory optionsAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().optionsAbs(absoluteURI)
    }

    @Signature
    default Memory optionsNow(Environment __env__, Memory... args) {
        this.getWrappedObject().optionsNow(options, responseHandler)
    }

    @Signature
    default Memory put(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().put(options)
    }

    @Signature
    default Memory putAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().putAbs(absoluteURI)
    }

    @Signature
    default Memory delete(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().delete(options)
    }

    @Signature
    default Memory deleteAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, HttpClientRequestImpl::new, this.getWrappedObject().deleteAbs(absoluteURI)
    }

    @Signature
    default Memory websocket(Environment __env__, Memory... args) {
        this.getWrappedObject().websocket(options, wsConnect)
    }

    @Signature
    default Memory websocketAbs(Environment __env__, Memory... args) {
        this.getWrappedObject().websocketAbs(url, headers, version, subProtocols, wsConnect, failureHandler)
    }

    @Signature
    default Memory websocketStream(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, ReadStreamImpl::new, this.getWrappedObject().websocketStream(options)
    }

    @Signature
    default Memory websocketStreamAbs(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, ReadStreamImpl::new, this.getWrappedObject().websocketStreamAbs(url, headers, version, subProtocols)
    }

    @Signature
    default Memory redirectHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().redirectHandler(handler)
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }

}
